lib/shimmer.js

```diff
@@ -648,3 +654,35 @@ function hasValidRegisterOptions(opts) {
 
   return true
 }
+
+function trackInstrumentationUsage(agent, shim, moduleName, metricPrefix) {
```

Comment:
mind adding some jsdoc here?  I also would love to see some unit tests for this code.  I do notice we have versioned tests that indirectly verify this but might be better for documentation and future selves

Reply:
Sure, I can add tests. We often don't for support metrics that aren't a part of a shared spec as these tend to be a bit more subject to change and removal but in this case we are looking to potentially leverage to drive decision making so makes sense to be a bit more careful.

jsdoc is fine as well although unlike to add much value.